lib/crowbar/client/command/proposal.rb

```diff
@@ -101,33 +104,160 @@ module Proposal
             parent.desc "Create a proposal for specific barclamp"
             parent.arg :barclamp
             parent.arg :proposal
-            parent.command :create do |c|
+            parent.command "create" do |c|
+              c.desc "Proposal data in json format"
+              c.flag [:data], type: String, default_value: nil
+
+              c.desc "Proposal data as a json file"
+              c.flag [:file], type: String, default_value: nil
+
+              c.desc "Merge input with proposal data"
+              c.switch [:merge], negatable: false
+
               c.action do |global, opts, args|
                 barclamp = args.shift
                 proposal = args.shift
                 helper.validate_availability_of! barclamp
 
-                raise "Not implemented yet! (#{barclamp}, #{proposal})"
+                template = Request.instance.proposal_template(
+                  barclamp
+                ).parsed_response.easy_merge(
+                  "id" => proposal
+                )
+
+                payload = case
+                when opts[:data]
+                  json = begin
+                    JSON.load opts[:data]
+                  rescue
+                    exit_now! "Invalid json data"
+                  end
+
+                  if opts[:merge]
+                    template.easy_merge json
+                  else
+                    json
+                  end
+                when opts[:file]
```

Comment:
Style/CaseIndentation: Indent `when` as deep as `case`. (https://github.com/bbatsov/ruby-style-guide#indent-when-to-case)